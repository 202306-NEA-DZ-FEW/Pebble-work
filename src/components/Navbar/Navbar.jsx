import Link from "next/link";
import { useTranslation } from "next-i18next";
import React, { useEffect, useState } from "react";

import styles from "@/styles/Navbar.module.css";
import { useRouter } from "next/router";

import { auth } from "@/util/firebase";

import Dropdown from "../Dropdown";
import Language from "../Language/Language";
import Pebble from "../Pebble";
import { motion } from "framer-motion";

let tabs = [
    { id: "", label: "Home" },
    { id: "events", label: "Events" },
    { id: "about", label: "About" },
];
const Navbar = () => {
    const [menuDropdownOpen, setMenuDropdownOpen] = useState(false);
    const [user, setUser] = useState(null);
    const [activeTab, setActiveTab] = useState(tabs[0].id);
    const { t } = useTranslation();
    const router = useRouter(); //so the motion will always stay on the current active tab

    const menuDropdown = () => {
        setMenuDropdownOpen(!menuDropdownOpen);
    };

    const closeMenuDropdown = () => {
        setMenuDropdownOpen(false);
    };

    useEffect(() => {
        const logged = auth.onAuthStateChanged((user) => {
            if (user) {
                setUser(user);
            } else {
                setUser(null);
            }
        });

        setActiveTab(router.pathname.slice(1)); //so the motion will always stay on the current active tab

        return () => {
            logged();
        };
    }, []);

    return (
        <nav className='sticky z-[9999] md:mb-10 mb-4 bg-[#B4CD93] top-0 xl:flex xl:flex-col xl:items-center'>
            <div
                style={{
                    width: "100%",
                    height: "8%",
                    zIndex: "9999",
                }}
                className='flex flex-wrap items-center justify-between p-4 xl:gap-40'
            >
                <Link href='./' className='flex items-center'>
                    <img
                        src='/logo/Logo.png'
                        className='h-8 mr-3'
                        alt='Pebble Logo'
                    />
                    <Pebble />
                </Link>
                <div className='flex items-center md:order-2'>
                    <div className='flex gap-4'>
                        <Dropdown />
                    </div>
                    <Language />
                    <button
                        data-collapse-toggle='navbar-language'
                        type='button'
                        onClick={menuDropdown}
                        className='inline-flex z-50 items-center p-2 ml-2 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600'
                        aria-controls='navbar-language'
                        aria-expanded='false'
                    >
                        <svg viewBox='0 0 100 80' width='20' height='20'>
                            <rect
                                fill='grey'
                                width='100'
                                height='15'
                                rx='10'
                            ></rect>
                            <rect
                                fill='grey'
                                y='25'
                                width='100'
                                height='15'
                                rx='10'
                            ></rect>
                            <rect
                                fill='grey'
                                y='50'
                                width='100'
                                height='15'
                                rx='10'
                            ></rect>
                        </svg>
                    </button>
                </div>
                <div
                    className={`${
                        menuDropdownOpen ? "block" : "hidden"
                    }  md:flex md:items-center w-full md:w-auto`}
                    id='navbar-language'
                >
                    <ul
                        className={`md:static text-center md:bg-transparent gap-10 fixed sm:w-full w-60 md:min-h-0 min-h-screen flex flex-col font-medium md:p-0 border border-gray-300 md:flex-row md:space-x-8 md:mt-0 md:border-0 dark:bg-gray-800 dark:border-gray-700 left-[0px] top-0 z-10 ${styles.tiltIn}`}
                        role='menu'
                    >
                        {tabs.map((tab) => (
                            <li key={tab.id}>
                                <Link href={`/${tab.id}`}>
                                    <p
                                        onClick={() => {
                                            setActiveTab(tab.id);
                                            closeMenuDropdown();
                                        }}
                                        className={`block rounded dark:border-gray-700 ${
                                            activeTab === tab.id
                                                ? "cursor-default" //text on hover while the motion is on it
                                                : "hover:text-[#547543]" //text on hover while the motion is not on it
                                        } relative rounded-full lg:text-[18px] md:text-[15px] font-medium text-white outline-sky-400 transition focus-visible:outline-2`}
                                        style={{
                                            WebkitTapHighlightColor:
                                                "transparent",
                                        }}
                                    >
                                        {activeTab === tab.id && (
                                            <motion.span
                                                layoutId='line'
                                                className='absolute bottom-0 left-0 right-0 h-1 z-10 bg-[#2E7EAA] mix-blend-difference'
                                                transition={{
                                                    type: "spring",
                                                    bounce: 0.3,
                                                    duration: 1.2,
                                                }}
                                            />
                                        )}
                                        {t(
                                            `common:navbar:${tab.label.toLowerCase()}`
                                        )}
                                    </p>
                                </Link>
                            </li>
                        ))}
                    </ul>
                </div>
            </div>
        </nav>
    );
};

export default Navbar;
