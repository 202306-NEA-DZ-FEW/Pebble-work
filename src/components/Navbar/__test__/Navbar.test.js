import React from "react";
import { render } from "@testing-library/react";
import Navbar from "../Navbar"; // adjust this import to your file structure

describe("Navbar", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<Navbar />);
        expect(asFragment()).toMatchSnapshot();
    });
});
