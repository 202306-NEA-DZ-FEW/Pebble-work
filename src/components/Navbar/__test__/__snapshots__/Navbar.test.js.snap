// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches the snapshot 1`] = `
<nav
  className="sticky z-[555] bg-[#B4CD93] top-0 xl:flex xl:flex-col xl:items-center"
>
  <div
    className="fixed -right-1 bottom-2"
  >
    <button
      className="expandingButton "
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height={15}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width={15}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M431 320.6c-1-3.6 1.2-8.6 3.3-12.2a33.68 33.68 0 012.1-3.1A162 162 0 00464 215c.3-92.2-77.5-167-173.7-167-83.9 0-153.9 57.1-170.3 132.9a160.7 160.7 0 00-3.7 34.2c0 92.3 74.8 169.1 171 169.1 15.3 0 35.9-4.6 47.2-7.7s22.5-7.2 25.4-8.3a26.44 26.44 0 019.3-1.7 26 26 0 0110.1 2l56.7 20.1a13.52 13.52 0 003.9 1 8 8 0 008-8 12.85 12.85 0 00-.5-2.7z"
          fill="none"
          strokeLinecap="round"
          strokeMiterlimit="10"
          strokeWidth="32"
        />
        <path
          d="M66.46 232a146.23 146.23 0 006.39 152.67c2.31 3.49 3.61 6.19 3.21 8s-11.93 61.87-11.93 61.87a8 8 0 002.71 7.68A8.17 8.17 0 0072 464a7.26 7.26 0 002.91-.6l56.21-22a15.7 15.7 0 0112 .2c18.94 7.38 39.88 12 60.83 12A159.21 159.21 0 00284 432.11"
          fill="none"
          strokeLinecap="round"
          strokeMiterlimit="10"
          strokeWidth="32"
        />
      </svg>
    </button>
    <div
      className="flex items-end absolute bottom-[100%] right-2 expandingDiv "
    >
      <div
        className="flex relative pl-2 justify-center max-h-full w-full scrollBar overflow-y-scroll"
      >
        <div
          className="w-full"
        >
          <div
            className="min-w-full"
            style={
              {
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <form
              onSubmit={[Function]}
            >
              <textarea
                className="min-w-full resize-none scrollBar"
                onChange={[Function]}
                onKeyDown={[Function]}
                style={
                  {
                    "height": "80px",
                    "paddingRight": "30px",
                  }
                }
                value=""
              />
              <button
                className="w-6"
                style={
                  {
                    "position": "absolute",
                    "right": "4px",
                    "top": "4px",
                  }
                }
                type="submit"
              >
                <img
                  alt="Send"
                  src="/icons/message.svg"
                />
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex flex-wrap items-center justify-between p-4 xl:gap-40"
    style={
      {
        "height": "8%",
        "width": "100%",
      }
    }
  >
    <a
      className="flex items-center"
      href="./"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <img
        alt="Pebble Logo"
        className="h-8 mr-3"
        src="/logo/Logo.png"
      />
      <div
        className="wave flex flex-row gap-1 lg:text-[18px] sm:text-[16px] text-[14px]"
      >
        <span
          className="letter"
          style={
            {
              "--i": 1,
            }
          }
        >
          P
        </span>
        <span
          className="letter"
          style={
            {
              "--i": 2,
            }
          }
        >
          E
        </span>
        <span
          className="letter"
          style={
            {
              "--i": 3,
            }
          }
        >
          B
        </span>
        <span
          className="letter"
          style={
            {
              "--i": 4,
            }
          }
        >
          B
        </span>
        <span
          className="letter"
          style={
            {
              "--i": 5,
            }
          }
        >
          L
        </span>
        <span
          className="letter"
          style={
            {
              "--i": 6,
            }
          }
        >
          E
        </span>
      </div>
    </a>
    <div
      className="flex items-center md:order-2"
    >
      <div
        className="flex gap-4"
      >
        <span
          style={
            {
              "display": "inherit",
            }
          }
        >
          <span
            style={
              {
                "animation": "react-spinners-BeatLoader-beat 0.7s 0s infinite linear",
                "animationFillMode": "both",
                "backgroundColor": "#749D60",
                "borderRadius": "100%",
                "display": "inline-block",
                "height": "10px",
                "margin": "2px",
                "width": "10px",
              }
            }
          />
          <span
            style={
              {
                "animation": "react-spinners-BeatLoader-beat 0.7s 0.35s infinite linear",
                "animationFillMode": "both",
                "backgroundColor": "#749D60",
                "borderRadius": "100%",
                "display": "inline-block",
                "height": "10px",
                "margin": "2px",
                "width": "10px",
              }
            }
          />
          <span
            style={
              {
                "animation": "react-spinners-BeatLoader-beat 0.7s 0s infinite linear",
                "animationFillMode": "both",
                "backgroundColor": "#749D60",
                "borderRadius": "100%",
                "display": "inline-block",
                "height": "10px",
                "margin": "2px",
                "width": "10px",
              }
            }
          />
        </span>
      </div>
      <div
        className="fixed md:text-[15px] text-[12px] bottom-0 left-20 sm:right-40 react-draggable"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        style={
          {
            "transform": "translate(0px,0px)",
          }
        }
        transform={null}
      >
        <button
          className="inline-flex items-center font-medium justify-center text-sm text-gray-900 rounded-lg cursor-pointer"
          onClick={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          <div
            className="flex items-center z-[444]"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-earth-europe fa-2x "
              data-icon="earth-europe"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                {
                  "color": "#2E7EAA",
                  "paddingRight": "2px",
                  "transform": "rotate(0deg)",
                  "transition": "transform 0.5s",
                }
              }
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M266.3 48.3L232.5 73.6c-5.4 4-8.5 10.4-8.5 17.1v9.1c0 6.8 5.5 12.3 12.3 12.3c2.4 0 4.8-.7 6.8-2.1l41.8-27.9c2-1.3 4.4-2.1 6.8-2.1h1c6.2 0 11.3 5.1 11.3 11.3c0 3-1.2 5.9-3.3 8l-19.9 19.9c-5.8 5.8-12.9 10.2-20.7 12.8l-26.5 8.8c-5.8 1.9-9.6 7.3-9.6 13.4c0 3.7-1.5 7.3-4.1 10l-17.9 17.9c-6.4 6.4-9.9 15-9.9 24v4.3c0 16.4 13.6 29.7 29.9 29.7c11 0 21.2-6.2 26.1-16l4-8.1c2.4-4.8 7.4-7.9 12.8-7.9c4.5 0 8.7 2.1 11.4 5.7l16.3 21.7c2.1 2.9 5.5 4.5 9.1 4.5c8.4 0 13.9-8.9 10.1-16.4l-1.1-2.3c-3.5-7 0-15.5 7.5-18l21.2-7.1c7.6-2.5 12.7-9.6 12.7-17.6c0-10.3 8.3-18.6 18.6-18.6H400c8.8 0 16 7.2 16 16s-7.2 16-16 16H379.3c-7.2 0-14.2 2.9-19.3 8l-4.7 4.7c-2.1 2.1-3.3 5-3.3 8c0 6.2 5.1 11.3 11.3 11.3h11.3c6 0 11.8 2.4 16 6.6l6.5 6.5c1.8 1.8 2.8 4.3 2.8 6.8s-1 5-2.8 6.8l-7.5 7.5C386 262 384 266.9 384 272s2 10 5.7 13.7L408 304c10.2 10.2 24.1 16 38.6 16H454c6.5-20.2 10-41.7 10-64c0-111.4-87.6-202.4-197.7-207.7zm172 307.9c-3.7-2.6-8.2-4.1-13-4.1c-6 0-11.8-2.4-16-6.6L396 332c-7.7-7.7-18-12-28.9-12c-9.7 0-19.2-3.5-26.6-9.8L314 287.4c-11.6-9.9-26.4-15.4-41.7-15.4H251.4c-12.6 0-25 3.7-35.5 10.7L188.5 301c-17.8 11.9-28.5 31.9-28.5 53.3v3.2c0 17 6.7 33.3 18.7 45.3l16 16c8.5 8.5 20 13.3 32 13.3H248c13.3 0 24 10.7 24 24c0 2.5 .4 5 1.1 7.3c71.3-5.8 132.5-47.6 165.2-107.2zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zM187.3 100.7c-6.2-6.2-16.4-6.2-22.6 0l-32 32c-6.2 6.2-6.2 16.4 0 22.6s16.4 6.2 22.6 0l32-32c6.2-6.2 6.2-16.4 0-22.6z"
                fill="currentColor"
                style={{}}
              />
            </svg>
            <p
              className="text-black md:text-[15px] text-[12px]"
            >
              common:language:language
            </p>
          </div>
        </button>
        <div
          className="z-50 translate-y-[200px] opacity-0 text-base list-none divide-y transform transition-all duration-700 divide-gray-100 fixed bottom-10 left-6 rounded-[4px] shadow bg-transparent"
          id="language-dropdown-menu"
        >
          <ul
            className="font-medium w-[100px] overflow-hidden flex flex-col gap-2 "
          >
            <li>
              <button
                className="block text-sm text-gray-700 hover:border hover:rounded-sm w-full dark:text-gray-400 dark:hover:bg-gray-600 dark:hover:text-white"
                onClick={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  className="inline-flex w-full items-center"
                >
                  <img
                    className="mr-1"
                    height="20px"
                    src="/United-Kingdom.svg"
                    width="20px"
                  />
                  common:language:english
                </div>
              </button>
            </li>
            <li>
              <button
                className="block text-sm text-gray-700 hover:border hover:rounded-sm w-full dark:text-gray-400 dark:hover:bg-gray-600 dark:hover:text-white"
                onClick={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  className="inline-flex w-full items-center"
                >
                  <img
                    className="mr-1"
                    height="20px"
                    src="/Turkey.svg"
                    width="20px"
                  />
                  common:language:turkish
                </div>
              </button>
            </li>
            <li>
              <button
                className="block text-sm text-gray-700 hover:border hover:rounded-sm w-full dark:text-gray-400 dark:hover:bg-gray-600 dark:hover:text-white"
                onClick={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  className="inline-flex w-full items-center"
                >
                  <img
                    className="mr-1"
                    height="20px"
                    src="/Germany.svg"
                    width="20px"
                  />
                  common:language:german
                </div>
              </button>
            </li>
            <li>
              <button
                className="block text-sm text-gray-700 hover:border hover:rounded-sm w-full dark:text-gray-400 dark:hover:bg-gray-600 dark:hover:text-white"
                onClick={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  className="inline-flex w-full items-center"
                >
                  <img
                    className="mr-1"
                    height="20px"
                    src="/China.svg"
                    width="20px"
                  />
                  common:language:chinese
                </div>
              </button>
            </li>
          </ul>
        </div>
      </div>
      <button
        aria-controls="navbar-language"
        aria-expanded="false"
        className="inline-flex z-50 items-center p-2 ml-2 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
        data-collapse-toggle="navbar-language"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="20"
          viewBox="0 0 100 80"
          width="20"
        >
          <rect
            fill="grey"
            height="15"
            rx="10"
            width="100"
          />
          <rect
            fill="grey"
            height="15"
            rx="10"
            width="100"
            y="25"
          />
          <rect
            fill="grey"
            height="15"
            rx="10"
            width="100"
            y="50"
          />
        </svg>
      </button>
    </div>
    <div
      className="hidden  md:flex md:items-center w-full md:w-auto navbar-content"
      id="navbar-language"
    >
      <ul
        className="md:static text-center md:bg-transparent gap-10 fixed sm:w-full w-60 md:min-h-0 min-h-screen flex flex-col font-medium md:p-0 md:flex-row md:space-x-8 md:mt-0 bg-gray-800 left-[0px] top-0 z-10 tiltIn"
        role="menu"
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <p
              className="block rounded dark:border-gray-700  relative rounded-full lg:text-[18px] md:text-[15px] font-medium text-white outline-sky-400 transition focus-visible:outline-2"
              onClick={[Function]}
              style={
                {
                  "WebkitTapHighlightColor": "transparent",
                }
              }
            >
              <span
                className="absolute bottom-0 left-0 right-0 md:h-[2px] h-1 z-10 bg-[#547543]"
                style={{}}
              />
              common:navbar:home
            </p>
          </a>
        </li>
        <li>
          <a
            href="/events"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <p
              className="block rounded dark:border-gray-700 hover:text-[#547543] relative rounded-full lg:text-[18px] md:text-[15px] font-medium text-white outline-sky-400 transition focus-visible:outline-2"
              onClick={[Function]}
              style={
                {
                  "WebkitTapHighlightColor": "transparent",
                }
              }
            >
              common:navbar:events
            </p>
          </a>
        </li>
        <li>
          <a
            href="/about"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <p
              className="block rounded dark:border-gray-700 hover:text-[#547543] relative rounded-full lg:text-[18px] md:text-[15px] font-medium text-white outline-sky-400 transition focus-visible:outline-2"
              onClick={[Function]}
              style={
                {
                  "WebkitTapHighlightColor": "transparent",
                }
              }
            >
              common:navbar:about
            </p>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
